### DISCUSSION TITLE
[E143] Labels for component 'tagger' not initialized on version 3.0.3

### DISCUSSION THREAD

**Discussion by HsengivS** [explosion/spaCy#7614] [Help: Coding & Implementations] [answered]
## My Environment

- Operating System: Windows 10
- Python Version Used: 3.7.10
- spaCy Version Used: 3.0.3
- Environment Information:

Hi 

I tried training a custom tagger, using the below command

`python -m spacy train config.cfg --output ./output --paths.train ./train.spacy --paths.dev ./dev.spacy`

I am getting the below error,

> ℹ Using CPU
> 
> =========================== Initializing pipeline ===========================
> Set up nlp object from config
> Pipeline: ['transformer', 'tagger']
> Created vocabulary
> Finished initializing nlp object
> Traceback (most recent call last):
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\runpy.py", line 193, in _run_module_as_main
>     "__main__", mod_spec)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\runpy.py", line 85, in _run_code
>     exec(code, run_globals)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\spacy\__main__.py", line 4, in <module>
>     setup_cli()
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\spacy\cli\_util.py", line 68, in setup_cli
>     command(prog_name=COMMAND)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\click\core.py", line 829, in __call__
>     return self.main(*args, **kwargs)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\click\core.py", line 782, in main
>     rv = self.invoke(ctx)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\click\core.py", line 1259, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\click\core.py", line 1066, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\click\core.py", line 610, in invoke
>     return callback(*args, **kwargs)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\typer\main.py", line 497, in wrapper
>     return callback(**use_params)  # type: ignore
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\spacy\cli\train.py", line 56, in train_cli
>     nlp = init_nlp(config, use_gpu=use_gpu)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\spacy\training\initialize.py", line 70, in init_nlp
>     nlp.initialize(lambda: train_corpus(nlp), sgd=optimizer)
>   File "C:\Users\SMVP\Anaconda3\envs\spacy_v3_gpu\lib\site-packages\spacy\language.py", line 1246, in initialize
>     proc.initialize(get_examples, nlp=self, **p_settings)
>   File "spacy\pipeline\tagger.pyx", line 270, in spacy.pipeline.tagger.Tagger.initialize
>   File "spacy\pipeline\pipe.pyx", line 104, in spacy.pipeline.pipe.Pipe._require_labels
> ValueError: [E143] Labels for component 'tagger' not initialized. This can be fixed by calling add_label, or by providing a representative batch of examples to the component's `initialize` method.

I can't find a way on how to add the custom labels.

Could anyone guide me on how this could be fixed...

Thanks in advance

**Comment by adrianeboyd** [comment, marked as answer]
The labels are normally read in from the docs `train.spacy` under `token.tag_`. Can you load the file and check that the docs are tagged?

```python
import spacy
from spacy.tokens import DocBin

nlp = spacy.blank("en")
db = DocBin().from_disk("train.spacy")
for doc in db.get_docs(nlp.vocab):
    assert doc.has_annotation("TAG")
    # or just inspect the tags
    for token in doc:
        print(token.text, token.tag_)
```

**Comment by Tayaress51** [comment]
Hi !

I'm pretty new to NLP.
I got the same problem so I inspect the tags in my train data.
There is no tags but I don't know how to get them.

I'm trying to made my NER indentifie aircraft models. I first think problem came from the custom NER tag "aircraft".
Problem come from POS tags if I understand correctly.

In my mind, tags for tagger where present per default.
Actually, my training data are very few (I'm testing), so I foud strange to create my POS tags with them.
 Will I use sample data like CoNLL before testing my training data ?

Please can you ligth my lantern ?
Thanks in advance.

**Reply by polm** [reply]
You shouldn't need to have POS tags in your training data in order to train an NER model. If you haven't done this before you might find it helpful to go through [the spaCy course](https://course.spacy.io/en/), which covers training an NER model. 

If you'd like more help with this, please open a new thread and include your config file.

**Reply by Tayaress51** [reply]
OK thanks !
I will follow the course and open an new thread if I got problems
